Compiling for 32 bit results in errors

I'm not sure if there's anything wrong in my setup however I get the following when I build, there are more similar errors from disk.rs module. 

```
     total_space = stat.f_bsize * stat.f_blocks;
   |                                          ^^^^^^^^^^^^^ expected u32, found u64
error[E0277]: the trait bound `u32: std::ops::Mul<u64>` is not satisfied
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,11 +10,27 @@ matrix:
       rust: stable
       dist: trusty
     - os: linux
+      env: TARGET=i686-unknown-linux-gnu
+      rust: stable
+      dist: trusty
+    - os: linux
+      env: TARGET=i686-unknown-linux-musl
+      rust: stable
+      dist: trusty
+    - os: linux
       env: TARGET=armv7-unknown-linux-gnueabihf
       rust: nightly
       dist: trusty
     - os: linux
       env: TARGET=x86_64-apple-darwin
+      rust: nightly
+      dist: trusty
+    - os: linux
+      env: TARGET=i686-unknown-linux-gnu
+      rust: nightly
+      dist: trusty
+    - os: linux
+      env: TARGET=i686-unknown-linux-musl
       rust: nightly
       dist: trusty
     - os: osx

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "sysinfo"
-version = "0.5.1"
+version = "0.5.2"
 authors = ["Guillaume Gomez <guillaume1.gomez@gmail.com>"]
 
 description = "Library to handle processes"

--- a/src/linux/component.rs
+++ b/src/linux/component.rs
@@ -104,10 +104,10 @@ impl Component {
                critical: Option<f32>) -> Component {
         let mut c = Component {
             temperature: 0f32,
-            label: label,
+            label,
             input_file: input_path.to_path_buf(),
             max: max.unwrap_or(0.0),
-            critical: critical,
+            critical,
         };
         c.update();
         c

--- a/src/linux/disk.rs
+++ b/src/linux/disk.rs
@@ -65,22 +65,22 @@ macro_rules! cast {
 pub fn new(name: &OsStr, mount_point: &Path, file_system: &[u8]) -> Disk {
     let mount_point_cpath = utils::to_cpath(mount_point);
     let type_ = find_type_for_name(name);
-    let mut total_space = 0;
-    let mut available_space = 0;
+    let mut total = 0;
+    let mut available = 0;
     unsafe {
         let mut stat: statvfs = mem::zeroed();
         if statvfs(mount_point_cpath.as_ptr() as *const _, &mut stat) == 0 {
-            total_space = stat.f_bsize * stat.f_blocks;
-            available_space = stat.f_bsize * stat.f_bavail;
+            total = stat.f_bsize * stat.f_blocks;
+            available = stat.f_bsize * stat.f_bavail;
         }
     }
     Disk {
-        type_: type_,
+        type_,
         name: name.to_owned(),
         file_system: file_system.to_owned(),
         mount_point: mount_point.to_owned(),
-        total_space: cast!(total_space),
-        available_space: cast!(available_space),
+        total_space: cast!(total),
+        available_space: cast!(available),
     }
 }
 

--- a/src/linux/disk.rs
+++ b/src/linux/disk.rs
@@ -52,12 +52,12 @@ fn find_type_for_name(name: &OsStr) -> DiskType {
     DiskType::from(rotational_int.unwrap_or(-1))
 }
 
-#[cfg(target_arch = "arm")]
+#[cfg(target_pointer_width = "32")]
 macro_rules! cast {
     ($x:expr) => { $x as u64 }
 }
 
-#[cfg(not(target_arch = "arm"))]
+#[cfg(not(target_pointer_width = "32"))]
 macro_rules! cast {
     ($x:expr) => { $x }
 }

--- a/src/linux/process.rs
+++ b/src/linux/process.rs
@@ -126,8 +126,8 @@ impl ProcessExt for Process {
     fn new(pid: Pid, parent: Option<Pid>, start_time: u64) -> Process {
         Process {
             name: String::new(),
-            pid: pid,
-            parent: parent,
+            pid,
+            parent,
             cmd: Vec::new(),
             environ: Vec::new(),
             exe: String::new(),
@@ -140,7 +140,7 @@ impl ProcessExt for Process {
             old_utime: 0,
             old_stime: 0,
             updated: true,
-            start_time: start_time,
+            start_time,
             uid: 0,
             gid: 0,
             status: ProcessStatus::Unknown(0),

--- a/src/linux/processor.rs
+++ b/src/linux/processor.rs
@@ -43,16 +43,16 @@ impl CpuValues {
                            irq: u64, softirq: u64, steal: u64, guest: u64,
                            guest_nice: u64) -> CpuValues {
         CpuValues {
-            user: user,
-            nice: nice,
-            system: system,
-            idle: idle,
-            iowait: iowait,
-            irq: irq,
-            softirq: softirq,
-            steal: steal,
-            guest: guest,
-            guest_nice: guest_nice
+            user,
+            nice,
+            system,
+            idle,
+            iowait,
+            irq,
+            softirq,
+            steal,
+            guest,
+            guest_nice,
         }
     }
 

--- a/src/mac/component.rs
+++ b/src/mac/component.rs
@@ -19,9 +19,9 @@ impl Component {
     pub fn new(label: String, max: Option<f32>, critical: Option<f32>) -> Component {
         Component {
             temperature: 0f32,
-            label: label,
+            label,
             max: max.unwrap_or(0.0),
-            critical: critical,
+            critical,
         }
     }
 }

--- a/src/mac/disk.rs
+++ b/src/mac/disk.rs
@@ -55,12 +55,12 @@ pub fn new(name: OsString, mount_point: &Path, type_: DiskType) -> Disk {
         }
     }
     Disk {
-        type_: type_,
-        name: name,
+        type_,
+        name,
         file_system: file_system.unwrap_or_else(|| b"<Unknown>".to_vec()),
         mount_point: mount_point.to_owned(),
-        total_space: total_space,
-        available_space: available_space,
+        total_space,
+        available_space,
     }
 }
 

--- a/src/mac/process.rs
+++ b/src/mac/process.rs
@@ -145,8 +145,8 @@ impl ProcessExt for Process {
     fn new(pid: Pid, parent: Option<Pid>, start_time: u64) -> Process {
         Process {
             name: String::new(),
-            pid: pid,
-            parent: parent,
+            pid,
+            parent,
             cmd: Vec::new(),
             environ: Vec::new(),
             exe: String::new(),
@@ -159,7 +159,7 @@ impl ProcessExt for Process {
             old_utime: 0,
             old_stime: 0,
             updated: true,
-            start_time: start_time,
+            start_time,
             uid: 0,
             gid: 0,
             process_status: ProcessStatus::Unknown(0),

--- a/src/mac/processor.rs
+++ b/src/mac/processor.rs
@@ -32,7 +32,7 @@ impl ProcessorData {
     pub fn new(cpu_info: *mut i32, num_cpu_info: u32) -> ProcessorData {
         ProcessorData {
             cpu_info: UnsafePtr(cpu_info),
-            num_cpu_info: num_cpu_info,
+            num_cpu_info,
         }
     }
 }
@@ -59,9 +59,9 @@ pub struct Processor {
 impl Processor {
     fn new(name: String, processor_data: Arc<ProcessorData>) -> Processor {
         Processor {
-            name: name,
+            name,
             cpu_usage: 0f32,
-            processor_data: processor_data,
+            processor_data,
         }
     }
 }

